app/scripts/controllers/membership.js

```diff
@@ -0,0 +1,304 @@
+            showAlert('currentUserLastRole', 'error', messages.notice.yourLastRole({roleName: roleName}), null, modalScope);
+          }
+        }
+        if(_.isEqual(kind, 'ServiceAccount')) {
+          if(_.startsWith(roleName, 'system:')) {
+            showAlert('editingServiceAccountRole', 'error', messages.warning.serviceAccount(), null, modalScope);
+          }
+        }
+        return modalScope;
+      };
+
+      ProjectsService
+        .get($routeParams.project)
+        .then(_.spread(function(project, context) {
+          requestContext = context;
+          refreshRoleBindingList();
+          angular.extend($scope, {
+            project: project,
+            subjectKinds: subjectKinds,
+            confirmRemove: function(subjectName, kindName, roleName) {
+              var redirectToProjectList = null;
+              var modalScope = createModalScope(subjectName, kindName, roleName, user.metadata.name);
+              if(_.isEqual(subjectName, user.metadata.name)) {
+                if(MembershipService.isLastRole(user.metadata.name, $scope.roleBindings)) {
+                  redirectToProjectList = true;
+                }
+              }
+              $uibModal.open({
+                animation: true,
+                templateUrl: 'views/modals/confirm.html',
+                controller: 'ConfirmModalController',
+                resolve: {
+                  modalConfig: function() {
+                    return modalScope;
+                  }
+                }
+              }).result.then(function() {
+                RoleBindingsService
+                  .removeSubject(subjectName, roleName, $scope.roleBindings, requestContext)
+                  .then(function() {
+                    if(redirectToProjectList) {
+                      $location.url("./");
+                    } else {
+                      refreshRoleBindingList();
+                      showAlert('rolebindingUpdate', 'success', messages.remove.success({
+                        roleName: roleName,
+                        subjectName: subjectName
+                      }));
+                    }
+                  }, function(err) {
+                    showAlert('rolebindingUpdateFail', 'error', messages.remove.error({
+                      roleName: roleName,
+                      subjectName: subjectName
+                    }),  messages.errorReason({
+                      httpErr: $filter('getErrorDetails')(err)
+                    }));
+                  });
+              });
+            },
+            addRoleTo:function(subjectName, kind, roleName, namespace) {
+              var subject = {
+                name: subjectName,
+                kind: kind,
+                namespace: namespace
+              };
+
+              var found = RoleBindingsService.findFirstByRole(roleName, $scope.roleBindings);
+              if(found) {
+                return updateBinding(found, subject, projectName);
```

Comment:
yeah this is where you have the issue that you can add a user to a role multiple times, the rolebinding already exists and update is just adding the subject to the list again